### PR TITLE
feat: consumable entitlements

### PR DESCRIPTION
**Pull request by almeidx**
**Please describe the changes this PR makes and why it should be merged:**
Added support for consumable entitlements

Upstream:
- https://github.com/discord/discord-api-docs/pull/6820

Depends on:
- https://github.com/discordjs/discord-api-types/pull/966
- #10238

**Status and versioning classification:**

### CODE REVIEW

File: packages/discord.js/src/structures/Entitlement.js [hunk 1]
```diff
@@ -159,6 +169,16 @@ class Entitlement extends Base {
...
     return this.client.users.fetch(this.userId);
   }
+
+  /**
+   * Marks this entitlement as consumed
+   * <info>Only available for One-Time Purchase consumable SKUs.</info>
+   * @returns {Promise<void>}
+   */
+  async consume() {
+    // Return this.client.rest.post(Routes.consumeEntitlement(this.applicationId, this.id));
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/Entitlement.js:179]: Is this still needed? Or was -types not yet updated with this 👀

Comment (Jiralite) [reply packages/discord.js/src/structures/Entitlement.js:179]: It is merged, but not yet released.

Comment (vladfrangu) [reply packages/discord.js/src/structures/Entitlement.js:179]: Oh. Want me to release it now so we can merge it in?

Comment (Jiralite) [reply packages/discord.js/src/structures/Entitlement.js:179]: Sure, why not. Famous last words...

Comment (vladfrangu) [reply packages/discord.js/src/structures/Entitlement.js:179]: Has been done

**Review by Jiralite — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

File: packages/discord.js/src/managers/EntitlementManager.js [hunk 2]
```diff
@@ -124,6 +124,19 @@ class EntitlementManager extends CachedManager {
...
     await this.client.rest.delete(Routes.entitlement(this.client.application.id, resolved));
   }
+
+  /**
+   * Marks an entitlement as consumed
+   * <info>Only available for One-Time Purchase consumable SKUs.</info>
+   * @param {EntitlementResolvable} entitlement The entitlement to consume
+   * @returns {Promise<void>}
+   */
+  async consume(entitlement) {
```
Comment (Jiralite) [inline packages/discord.js/src/managers/EntitlementManager.js:134]: Can we keep this at a low level and only require the id?

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**